### ISSUE TITLE
Pool Interface

### ISSUE THREAD

**Issue by kevinconway** [kevinconway/scriptpool#2]
A `pool` is a resource that acts as an interface and manager for the underlying `worker` resources.

The responsibility of a `pool` is:
- Create and stop `worker` resources.
- Listen for messages intended for `worker` resources.
- Allocate messages to `worker` resources.

The proposed interface for the `pool` is:

> Pool [arguments]
> 
> arguments:
> 
>   --workers:        Number of workers to make (default 1).
>   --terminator:     Stop command (default 'terminate_pool').
>   --poll:           Idle wait time for a worker when queue is empty (default .1 seconds).

Each implementation must expose this public interface. Additional arguments specific to the implementation may be added, but they must not be necessary for starting a functioning pool. All implementation should have reasonable defaults for their arguments such that a call built using the above interface will successfully create a pool (even if not an optimal one).

In other words, the command `Pool` with no arguments should create a functional pool.

The proposed first implementations of this interface are:
- NamedPipePool
  
  A pool that listens for messages over a named pipe.
- RabbitPool
  
  A pool that listens for messages over RabbitMQ.
- SockPool
  
  A pool that listens for messages over a network socket.

Pool implementations should make use of the Worker and Queue interfaces.
